### PR TITLE
Allow to benchmark Elasticsearch with and without x-pack

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
x-pack, which has been previously implemented as a (closed-source) plugin, has
been recently opened. During that process, x-pack turned from a plugin into a
core module of Elasticsearch. Furthermore, there are now two distributions:

* The "default" one (with x-pack) 
* An OSS one (without x-pack)

This has several implications on our benchmarks as well. With this commit we
implement the following changes:

* x-pack is now treated as a "car" in Rally (as it is now a module and not a
plugin anymore). Strictly speaking, this change has happened in rally-teams
because Rally does not know about individual plugins. 
* The Gradle build commands are now read from properties in rally-teams
* Rally can now call bootstrapping code from rally-teams for cars as well
(previously this was only possible for plugins). One restriction is that we only
allow to load one bootstrap handler.

### CODE REVIEW

File: esrally/mechanic/supplier.py [hunk 1]
```diff
@@ -173,28 +172,35 @@ def __call__(self, *args, **kwargs):
...
-            return glob.glob("%s/distribution/archives/tar/build/distributions/*.tar.gz" % self.src_dir)[0]
+            return glob.glob("{}/{}".format(self.src_dir, self.value_of("artifact_path_pattern")))[0]
         except IndexError:
             raise SystemSetupError("Couldn't find a tar.gz distribution. Please run Rally with the pipeline 'from-sources-complete'.")
 
+    def value_of(self, k):
+        try:
+            return self.car.variables[k]
+        except KeyError:
+            raise exceptions.SystemSetupError("Car '{}' misses config variable '{}' to build Elasticsearch.".format(self.car, k))
```
Comment (dliappis) [inline esrally/mechanic/supplier.py]: maybe `is missing` is clearer here

File: esrally/mechanic/team.py [hunk 2]
```diff
@@ -380,24 +459,25 @@ def load(self):
...
-                logger.info("Invoking hook [%s].", hook.__name__)
+                logger.info("Invoking bootstrap hook [%s].", hook.__name__)
                 # hooks should only take keyword arguments to be forwards compatible with Rally!
-                hook(config_names=self.plugin.config, **kwargs)
+                hook(config_names=self.component.config, **kwargs)
         else:
-            logger.debug("Plugin [%s] in config [%s] has no hook registered for phase [%s].", self.plugin.name, self.plugin.config, phase)
\ No newline at end of file
+            logger.debug("Component [%s] in config [%s] has no hook registered for phase [%s].",
+                         self.component.name, self.component.config, phase)
```
Comment (dliappis) [inline esrally/mechanic/team.py]: Newline missing as pointed by GitHub

File: tests/mechanic/data/cars/v1/vanilla/config.ini [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[variables]
+# this should *always* be overridden by the car definition (config base
+# variables take least precedence)
+heap_size=0
+clean_command=./gradlew clean
```
Comment (dliappis) [inline tests/mechanic/data/cars/v1/vanilla/config.ini]: Same newline missing comment

File: tests/mechanic/data/cars/v1/verbose_logging/config.ini [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[variables]
+verbose_logging=true
```
Comment (dliappis) [inline tests/mechanic/data/cars/v1/verbose_logging/config.ini]: Same newline missing comment

File: esrally/mechanic/team.py [hunk 5]
```diff
@@ -380,24 +391,25 @@ def load(self):
...
```
Comment (dliappis) [inline esrally/mechanic/team.py]: Newline missing here

File: esrally/mechanic/supplier.py (hunk 1 above)
Comment (danielmitterdorfer) [reply esrally/mechanic/supplier.py]: Agreed. I changed it.

File: esrally/mechanic/team.py (hunk 2 above)
Comment (danielmitterdorfer) [reply esrally/mechanic/team.py]: Fixed.

File: tests/mechanic/data/cars/v1/vanilla/config.ini (hunk 3 above)
Comment (danielmitterdorfer) [reply tests/mechanic/data/cars/v1/vanilla/config.ini]: Fixed

File: tests/mechanic/data/cars/v1/verbose_logging/config.ini (hunk 4 above)
Comment (danielmitterdorfer) [reply tests/mechanic/data/cars/v1/verbose_logging/config.ini]: Fixed

**Pull request merged**